### ISSUE TITLE
Macro escape! leaks unused variable 'e' when 'verbose-errors' feature is disabled

### ISSUE THREAD

**Issue by willem66745** [Geal/nom#379]
After upgrading my projects to nom 2.0 the `escape!` macro gave the next error:

```
warning: unused variable: `e`, #[warn(unused_variables)] on by default
   --> src\const.rs:127:1
    |
127 | named!(const_value_string<ConstValue>,
    | ^
    |
    = note: this error originates in a macro outside of the current crate
```

Because almost identical code did give different results, it turns out that the `verbose-errors` removes the warning.

Looks like that the next bit in the `escape!` macro causes this problem:

```rust
      match cl() {
        $crate::IResult::Incomplete(x) => $crate::IResult::Incomplete(x),
        $crate::IResult::Done(i, o)    => $crate::IResult::Done(i, o),
        $crate::IResult::Error(e)      => {
          return $crate::IResult::Error(error_node_position!($crate::ErrorKind::Escaped, $i, e))
        }
      }
```

When the verbose errors are disabled, `e` is never used.

**Comment by jpastuszek** [comment]
Same here:
```
warning: unused variable: `e`, #[warn(unused_variables)] on by default
  --> src/input/syslog.rs:20:1
   |
20 | named!(priority<&[u8], u8>, return_error!(ErrorKind::Custom(1),
   | ^
   |
   = note: this error originates in a macro outside of the current crate
```

**Comment by awsaba** [comment]
The same happens for the `switch!` macro as well.

**Comment by porglezomp** [comment]
I tried to find a fix, it looks like simply doing
```rust
macro_rules! error_node_position(
  ($code:expr, $input: expr, $next:expr) => ({ let _ = $next; $code });
);
```
stops it producing a warning. This shouldn't adversely affect code generation in any way, correct?
Should I submit a PR with this?

**Comment by Geal** [comment]
@porglezomp this looks tempting, but I just tested: it makes type inference errors in some parsers, so it will not work :/
Until I find a better solution, I'll add a `#[allow(unused_variables)]` in `named!`.

**Comment by Geal** [comment]
fixed in 20d4670.
@porglezomp if you find a better solution, I'd be happy to take a PR :)

**Comment by 0xd34d10cc** [comment]
Looks like `named_args!` generates same warnings:
```
named_args!(parser(argument: u32)<u32>,
    switch!(value!(argument),
        1 => value!(2) |
        2 => value!(3)
    )
);
```

**Comment by Geal** [comment]
alright, this is fixed with d7d91d2 now :)

**Comment by Nervengift** [comment]
you seem to still get this error in custom functions

```rust
fn parse(i: &[u8], arg: u32) -> IResult<&[u8], u32> {
    switch!(i, value!(arg),
        0 => value!(0) |
        1 => value!(1)
    )
}
```

**Comment by Nervengift** [comment]
oh, my bad, the fix is not in 3.2, is it?
